src/blob.rs

```diff
@@ -39,6 +39,27 @@ impl Blob {
         }
     }
 
+    /// Creates a new `Blob` from the provided byte slice and assumes it's
+    /// already padded according to DA specs.
+    pub fn from_padded_bytes(input: &[u8]) -> Result<Self, BlobError> {
+        // check to see if bytes are modulo bn254
+        // set_bytes_canonical used in to_fr_array calls from_be_bytes_mod_order
+        // if the bytes passed into set_bytes_canonical are larger than the bn254 field
+        // modulo order then the bytes will be modded by the order of the field
+        let length_after_padding = input.len();
+        let fr_vec = helpers::to_fr_array(input);
+        let bytes = helpers::to_byte_array(&fr_vec, length_after_padding);
```

Comment:
don't think this conversion and check is needed. Better to add tests with random lengths to make sure the helper functions are working as expected.

Reply:
Not sure what a better way to check to make sure the byte array conforms to the padding spec is - I think adding a disclaimer that if any of the 32 byte elements in the input array is greater than the field modulo then this will modify the input is also fine as an option though - lmk what you think

Reply:
discussed offline with @anupsv , will remove this conversion and add a disclaimer that if the input data doesn't conform to the da spec it will be transformed.

Reply:
b10dbff6a5391d4ef30acb8dcbe8538f199bd2af